src/screens/FilterScreen.js

```diff
@@ -1,34 +1,198 @@
 
+import {CheckBox, Button, Icon} from 'react-native-elements';
+import Accordion from 'react-native-collapsible/Accordion';
 
-import {CardSection, Card} from '../components/common/index';
-import MapScreen from "../components/SearchMap";
+import {checkboxWasClicked} from "../actions";
+
+import {navKeys} from '../constants';
+import {campsite, more_screen, reducerAlerts} from '../locale.en';
+import {campsiteIcon, grey} from "../styles";
+
+const {campsite_form: {accessibility, facilities, price, accessibility_options, facilities_options, price_options}} = campsite;
+
+
+const ACCESSIBILITY = [
+    {
+        title: accessibility,
+        content: accessibility_options
+    }
+];
+
+const FACILITIES = [
+    {
+        title: facilities,
+        content: facilities_options
+    }
+];
+
+const PRICE = [
+    {
+        title: price,
+        content: price_options
+    }
+];
 
 class FilterScreen extends Component {
 
+    componentWillMount() {
+        const {displaySites} = this.props;
+
+        this.props.navigation.setParams({siteCount: displaySites.length});
+    }
+
+    componentWillReceiveProps(nextProps) {
+        const nextSetOfSites = nextProps.displaySites;
+        const currentSites = this.props.displaySites;
+
+
+        if (nextSetOfSites.length !== currentSites.length) {
+            this.props.navigation.setParams({siteCount: nextSetOfSites.length});
+        }
+    }
+
+    static renderRightNavButton = (navigate, {siteCount}) => {
+        if (Platform.OS === 'ios') {
+            return (
+                <Button
+                    title={_.isUndefined(siteCount) ? '' : `${siteCount} Results`}
+                    onPress={() => navigate(navKeys.SEARCH)}
+                    backgroundColor="rgba(0,0,0,0)"
+                    color="rgba(0,122,255,1)"
+                />
+            );
+        } else if (Platform.OS === 'android') {
+            // android-specific code for navigation here
+        }
+    };
+
     static navigationOptions = (props) => {
-        const {navigation: {navigate}} = props;
+        const {navigation: {navigate, state: {params = {}}}} = props;
 
         return {
-            headerTitle: 'Filter Your Search'
+            headerRight: FilterScreen.renderRightNavButton(navigate, params)
         }
 
     };
 
+    renderHeader = (section, index, isActive) => {
+        const {headerStyle, headerTextStyle} = styles;
 
-    render(){
-        return(
-            <Card>
-                <CardSection>
-                    <Text>
-                        Filter your sites
-                    </Text>
-                </CardSection>
-            </Card>
+        return (
+            <View style={headerStyle}>
+                <Text style={headerTextStyle}>
+                    {section.title}
+
+                </Text>
+                <Icon type='entypo' name={isActive ? 'chevron-down' : 'chevron-up'} size={25} color={campsiteIcon}/>
+            </View>
+        );
```

Comment:
Above is the render logic for the collapsible's header. Using the `isActive` prop that comes with `react-native-collapsible` we can do some fun logic with the icon we show.